src/openforms/contrib/haal_centraal/tests/test_integration.py

```diff
@@ -0,0 +1,147 @@
+    The prefill feature is being used here, although any other feature making use of the Haal Centraal API
+    could have been used.
+    """
+
+    VCR_TEST_FILES = Path(__file__).parent / "files"
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        global_config = GlobalConfiguration(organization_oin="00000003273229750000")
+
+        global_config_patcher = patch(
+            "openforms.contrib.haal_centraal.clients.GlobalConfiguration.get_solo",
+            return_value=global_config,
+        )
+        global_config_patcher.start()
+        self.addCleanup(global_config_patcher.stop)
+
+        form_step = FormStepFactory.create(form_definition__configuration=CONFIGURATION)
+        self.submission_step = SubmissionStepFactory.create(
+            submission__form=form_step.form, form_step=form_step
+        )
+
+        store_auth_details(
+            submission=self.submission_step.submission,
+            form_auth={
+                "attribute": AuthAttribute.bsn,
+                "plugin": "demo",
+                "value": "999990421",
+            },
+        )
+        store_registrator_details(
+            submission=self.submission_step.submission,
+            registrator_auth={
+                "attribute": AuthAttribute.employee_id,
+                "plugin": "dummy",
+                "value": "SAP001",
+            },
+        )
+
+    def validate_prefill_variables(self, value: JSONValue) -> None:
+        self.assertEqual(
+            value,
+            {
+                "aanhef": "Geachte heer Holthuizen",
+                "aanschrijfwijze": {"naam": "A.H. Holthuizen"},
+                "gebruikInLopendeTekst": "de heer Holthuizen",
+                "adresregel1": "Nieuwe Parklaan 58",
+                "adresregel2": "2597 LD  'S-GRAVENHAGE",
+            },
+        )
+
+    @patch(
+        "openforms.contrib.haal_centraal.clients.HaalCentraalConfig.get_solo",
+        return_value=HaalCentraalConfigFactory.build(
+            default_x_doelbinding_header="BRPACT-AanschrijvenZakelijkGerechtigde",
+            default_x_verwerking_header="Financiële administratie@Correspondentie factuur",
+        ),
+    )
+    def test_brp_integration_prefill_default_headers(self, m: MagicMock) -> None:
+        # certificate instances need to be saved so that they get a correct path
+        # (see https://forum.djangoproject.com/t/7533/)
+        if (
+            m.return_value.brp_personen_service.client_certificate
+            and m.return_value.brp_personen_service.server_certificate
+        ):
+            m.return_value.brp_personen_service.client_certificate.save()
+            m.return_value.brp_personen_service.server_certificate.save()
+
+        prefill_variables(submission=self.submission_step.submission)
+
+        state = self.submission_step.submission.load_submission_value_variables_state()
+        self.validate_prefill_variables(state.variables["address"].value)
```

Comment:
when you're writing custom assertions, call them as such: `self.assertPrefillVariableValues`

In a test it must be visually clear that you're doing:

```
setup phase

do work phase

assertion phase
```
so please also add those blank lines in your test code to make these blocks visually clear